pysqldb3/geopackage.py

```diff
@@ -374,90 +428,42 @@ def read_gpkg(self, dbo = None, schema = None, table = None, port = 5432, srid =
-        if not self.gpkg_tbl:
 
-            #######
-            # if no table name listed in argument, then we assume bulk upload
-            # this prompts us to count the number of tables within the geopackage to confirm bulk upload
-            
-            # confirm number of tables within the geotable
-            count_cmd = COUNT_GPKG_LAYERS.format(full_path = full_path) 
+        if bulk_upload == False:
 
-            try:
-                ogr_response = subprocess.check_output(shlex.split(count_cmd.replace('\n', ' ')), stderr=subprocess.STDOUT)
-                tables_in_gpkg = re.findall(r"\\n\d+:\s(?:[a-z0-9A-Z_]+)", str(ogr_response)) # only allows tables names with underscores, numbers, and letters
-                tbl_count = len(tables_in_gpkg)
-            
-            except subprocess.CalledProcessError as e:
-                print("Ogr2ogr Output:\n", e.output)
-                print('Ogr2ogr command failed. The Geopackage was not read in.')
-                raise subprocess.CalledProcessError(cmd=print_cmd_string([dbo.password], count_cmd), returncode=1)
-            
-            if tbl_count > 1:
-
-                assert not table, """Since the geopackage has more than 1 layer to be read, you cannot specify one output table name to apply to all the tables. \n
-                                    Either run the function for each table with customized output table names separately, \n
-                                    or remove the 'db_table' or 'table' argument for bulk upload.
-                                    """
-
-            # create a list of cleaned gpkg table names
-                gpkg_tbl_names = {} # create empty dictionary
-                for t_i_g in tables_in_gpkg:
-                    insert_val = re.search(r'([a-z0-9A-Z_]+)$', t_i_g).group()
-                    gpkg_tbl_names[insert_val] = insert_val # add the cleaned name
-
-            # if the bulk upload involves only 1 table and no db table name was specified, we name the table the same thing as the original gpkg table
-            elif tbl_count == 1:            
-                gpkg_tbl_names = {re.search(r'([a-z0-9A-Z_]+)$', tables_in_gpkg[0]).group(): table}
-
-        else:
-
-            if not table:
-                table = self.gpkg_tbl.replace('.gpkg', '').lower()
-
-            # create empty list of table names that will get read into the db
-            gpkg_tbl_names = {self.gpkg_tbl: table}
+            if not gpkg_tbl:
+                gpkg_tbl = self.gpkg_tbl
 
+            if table:
+                assert table == re.sub(r'[^A-Za-z0-9_]+', r'', table) # make sure the name will load into the database
+                table = table.lower()
+            else:
+                # clean the geopackage table name
+                table = gpkg_tbl.replace('.gpkg', '').lower()
+                table = re.sub(r'[^A-Za-z0-9_]+', r'', table) # clean the table name in case there are special characters
```

Comment:
you could wrap the table names in `"` so the special characters would be allowed, other wise if you have a table named `tbl_1` and `tbl_1*` you will get a collision in the database